src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/FileUpload/FileUploadWithTag/FileUploadWithTagComponent.tsx

```diff
@@ -0,0 +1,257 @@
+
+export function FileUploadWithTagComponent(props: IFileUploadWithTagProps): JSX.Element {
+  const dataDispatch = useDispatch();
+  const [validations, setValidations] = React.useState<Array<{ id: string, message: string }>>([]);
+  const mobileView = useMediaQuery('(max-width:992px)'); // breakpoint on altinn-modal
+  const options = useSelector((state: IRuntimeState) => state.optionState.options[props.optionsId]);
+  // eslint-disable-next-line max-len
+  const editIndex = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.fileUploadersWithTag[props.id]?.editIndex ?? -1);
+  // eslint-disable-next-line max-len
+  const chosenOptions = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.fileUploadersWithTag[props.id]?.chosenOptions ?? {});
+
+  const attachments: IAttachment[] = useSelector(
+    (state: IRuntimeState) => state.attachments.attachments[props.id] || emptyArray,
+  );
+
+  const setValidationsFromArray = (validationArray: string[]) => {
+    setValidations(
+      validations.concat(parseFileUploadComponentWithTagValidationObject(validationArray)),
+    );
+  };
+
+  const setEditIndex = (index: number) => {
+    dataDispatch(FormLayoutActions.updateFileUploaderWithTagEditIndex({
+      uploader: props.id, index
+    }));
+  };
+
+  const onClick = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+    event.preventDefault();
+  };
+
+  const onClickEdit = (index) => {
+    if (editIndex === -1 || editIndex !== index) {
+      setEditIndex(index);
+    } else {
+      setEditIndex(-1);
+    }
+  };
+
+  const onClickSave = (attachment: IAttachment) => {
+    if (chosenOptions[attachment.id] !== undefined && chosenOptions[attachment.id].length !== 0) {
+      setEditIndex(-1);
+      if (attachment.tags === undefined || chosenOptions[attachment.id] !== attachment.tags[0]) {
+        setAttachmentTag(attachment, chosenOptions[attachment.id]);
+      }
+      setValidations(validations.filter((obj) => obj.id !== attachment.id)); // Remove old validation if exists
+    } else {
+      const tmpValidations: { id: string, message: string }[] = [];
+      tmpValidations.push(
+        {
+          id: attachment.id,
+          message: `${getLanguageFromKey('form_filler.file_uploader_validation_error_no_chosen_tag', props.language)} ${props.getTextResource(props.textResourceBindings.tagTitle).toLowerCase()}.`,
+        },
+      );
+      setValidations(validations.filter((obj) => obj.id !== tmpValidations[0].id).concat(tmpValidations));
+    }
+  };
+
+  const onDropdownDataChange = (id: string, value: string) => {
+    if (value !== undefined) {
+      const option = options?.find((o) => o.value === value);
+      if (option !== undefined) {
+        dataDispatch(FormLayoutActions.updateFileUploaderWithTagChosenOptions({
+          uploader: props.id, id, option
+        }));
+      } else console.error(`Could not find option for ${value}`);
+    } else console.error('Should not be called');
```

Comment:
I would suggest to avoid inline if/else like this. Its easy to make an error here when refactoring, if this area is not covered enough by tests. Adding the extra {} helps avoid potential simple refactor issues. Its also very easy to miss these inline statements when reading through the code.

```suggestion
      } else {
        console.error(`Could not find option for ${value}`);
      }
    } else {
      console.error('Should not be called');
    }
```

Btw, is there any need to log to console `Should not be called` ?

Reply:
Should be no need, removed.